File: 900/1543A/main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() 
{
    int t;
    cin >> t;

    while (t--) 
    {
        long long a, b;
        cin >> a >> b;

        long long c = abs(a - b);
        cout << c;

        if (a == b)
            cout << " " << 0 << "\n";
        else
            cout << " " << min(a % c, c - a % c) << "\n";
    }

    return 0;
}

File: 900/1543A/main.java
import java.util.*;
public class main
{
    public static void main(String args[])
    {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        
        while(t-- > 0)
        {
            long a = sc.nextLong();
            long b = sc.nextLong();
            long c = Math.abs(a-b);
            System.out.print(c);
            
            if(a == b)
                System.out.println(" " + 0);
            else
                System.out.println(" " + Math.min(a%c, c - a%c));
        }
    }
}